Sources/SE0000_KeyPathReflection/Shims.swift

```diff
@@ -10,14 +10,15 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef KEYPATH_REFLECTION_FUNCTIONS
-#define KEYPATH_REFLECTION_FUNCTIONS
+// This file defines shims for runtime entry points.
 
-#include <stddef.h>
+import Swift
 
-// Used to allocate keypaths at runtime.
-
-// HeapObject *swift_allocObject(Metadata *type, size_t size, size_t alignMask);
-extern void *swift_allocObject(void *type, size_t size, size_t alignMask);
-
-#endif
+// SWIFT_RUNTIME_EXPORT
+// HeapObject *(*SWIFT_RT_DECLARE_ENTRY _swift_allocObject)(
+//     HeapMetadata const *metadata, size_t requiredSize,
+//     size_t requiredAlignmentMask) = _swift_allocObject_;
+@_silgen_name("swift_allocObject")
+func swift_allocObject<T: AnyObject>(
+  _ type: T.Type, requiredSize: Int, requiredAlignmentMask: Int
+) -> T
```

Comment:
The issue with this is that the `swift_allocObject` function is declared using the C calling convention which is why I used the CShim target to get it correctly bridged into Swift. I know Joe makes a point of this often because using `@_silgen_name` like this imports the function using the Swift calling convention which doesn't always align with the C one. Maybe the register usages of this function is the same for both calling conventions which is  why it works, but I wouldn't want to risk it.

Reply:
Thanks for the context. I've reverted this.